lib/kitchen/directions/bake_generic_eoc_section/v1.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Kitchen::Directions::BakeGenericEocSection
+  class V1
+    renderable
+
+    def bake(chapter:, metadata_source:, klass:, append_to: nil, uuid_prefix: '.',
+             include_intro: true, &block)
+      section_clipboard = Kitchen::Clipboard.new
+      sections = if include_intro
+                   chapter.pages.search("$.#{klass}")
+                 else
+                   chapter.non_introduction_pages.search("$.#{klass}")
```

Comment:
IIRC using a `$` here doesn't do anything with a generic `search` call.  A place you would use a `$` is if you're trying to modify the selector for a specialized search call, like instead of `chapter.pages` you could say `chapter.pages("$.#{klass}")` which is the same as `chapter.search("div[data-type='page'].#{klass}")`.

Relatedly, this search as is will be without a tag name so it will be slower.  Should we change `klass` to `section_selector` and have callers pass in e.g. `'section.key-equations'`?  Such a name change in conjuction with renaming the direction would do a lot to clear up the purpose of the direction and what its variables mean... `MoveSectionsToEoc.v1(... section_selector: 'section.key-equations' ...)` makes it pretty obvious that you are moving `section.key-equations` elements to the end of the chapter.

Reply:
Interesting ... is the `$` just a concatenation character then?

I have previously experimented with including the `section` element name in the selector. Here's what I've tried:

1. Originally, I had it pass in `klass` and then search with `.search('section.klass')`. This changed because @LostMojo mentioned some books have `div`s instead of `section`s. 
2. Next, I tried a call to pass in `selector` instead of `klass`, intending for something like `section.klass` to be given. But this would need some string parsing, because, as you pointed out above, the `klass` is used more than just as a CSS selector for the sections -- it's also used as a title lookup key and as part of the classes on the wrapper. So I would have to break the `klass` portion out of `section.klass` at some point. This is definitely doable, but since a `selector` parameter doesn't necessarily indicate an exclusively `name.class` format, I thought it would be better to specify what's expected. 

I can switch the implementation to 2) though, since I agree it would be faster. The name `section_selector` may make it more clear that the selector has an expected format. What do you think?

Reply:
Yes on (2) and as you saw below recommending we don't reuse this field for the other purposes, so string parsing won't be necessary

Reply:
The `$` character, when passed to a specific enumerator (like `pages`, not a general enumerator like `search`) gets replaced with the CSS for that specific enumerator, in the case of pages it gets replaced with `div[data-type="page"]`